projects/components/src/datagrid/datagrid.component.ts

```diff
@@ -274,6 +261,41 @@ export class DatagridComponent<R> implements OnInit {
     @Input() trackBy: TrackByFunction<R> = (index: number, record: (R & HasHref) | undefined): string | number =>
         record && (record.href || index)
 
+    /**
+     * Returns the buttons that should be featured given the {@param selection} of currently selected entities.
+     */
+    getFeaturedButtons(selection: R[]): ContextualButton<R>[] {
+        let toTake = this.buttonConfig.contextualButtons.featuredCount;
```

Comment:
This would work, except for the logic that if a featured button is disabled, we want to skip it and find another. Because of this, we might need to check more `featured` buttons than just `.slice(0, featuredButtonCount)` would give.